core/src/main/java/brooklyn/catalog/internal/BasicBrooklynCatalog.java

```diff
@@ -491,43 +508,127 @@ public void load() {
 
-        Collection<CatalogBundle> libraries = Collections.emptyList();
-        Maybe<Object> possibleLibraries = catalog.getMaybe("libraries");
-        if (possibleLibraries.isAbsent()) possibleLibraries = catalog.getMaybe("brooklyn.libraries");
-        if (possibleLibraries.isPresentAndNonNull()) {
-            if (!(possibleLibraries.get() instanceof Collection))
-                throw new IllegalArgumentException("Libraries should be a list, not "+possibleLibraries.get());
-            libraries = CatalogItemDtoAbstract.parseLibraries((Collection<?>) possibleLibraries.get());
+        if (sourceYaml==null) sourceYaml = new Yaml().dump(itemMetadata);
+
+        Map<Object,Object> catalogMetadata = MutableMap.builder().putAll(parentMetadata).putAll(itemMetadata).build();
+        
+        // libraries we treat specially, to append the list, with the child's list preferred in classloading order
+        List<?> librariesL = MutableList.copyOf(getFirstAs(itemMetadata, List.class, "brooklyn.libraries", "libraries").orNull())
+            .appendAll(getFirstAs(parentMetadata, List.class, "brooklyn.libraries", "libraries").orNull());
+        if (!librariesL.isEmpty())
+            catalogMetadata.put("brooklyn.libraries", librariesL);
+        Collection<CatalogBundle> libraries = CatalogItemDtoAbstract.parseLibraries(librariesL);
+
+        Object items = catalogMetadata.remove("items");
+        Object item = catalogMetadata.remove("item");
+
+        if (items!=null) {
+            int count = 0;
+            for (Map<?,?> i: ((List<Map<?,?>>)items)) {
+                addAbstractCatalogItems(Yamls.getTextOfYamlAtPath(sourceYaml, "items", count).getMatchedYamlTextOrWarn(), 
+                    i, result, catalogMetadata, whenAddingAsDtoShouldWeForce);
+                count++;
+            }
         }
+        
+        if (item==null) return;
 
-        final String id = (String) catalog.getMaybe("id").orNull();
-        final String version = Strings.toString(catalog.getMaybe("version").orNull());
-        final String symbolicName = (String) catalog.getMaybe("symbolicName").orNull();
-        final String name = (String) catalog.getMaybe("name").orNull();
-        final String displayName = (String) catalog.getMaybe("displayName").orNull();
-        final String description = (String) catalog.getMaybe("description").orNull();
-        final String iconUrl = (String) catalog.getMaybe("iconUrl").orNull();
-        final String iconUrlUnderscore = (String) catalog.getMaybe("icon_url").orNull();
-        final String deprecated = (String) catalog.getMaybe("deprecated").orNull();
+        // now look at the actual item, first correcting the sourceYaml and interpreting the catalog metadata
+        String itemYaml = Yamls.getTextOfYamlAtPath(sourceYaml, "item").getMatchedYamlTextOrWarn();
```

Comment:
You are keeping track of three views on the same markup: 
  * the string (sub-)yaml
  * the parsed map
  * the camp plan

Why not deal with just the parsed YAML and pass it directly to pdp parser? This will greatly simplify the code. The only place where `Yamls.getTextOfYamlAtPath` is really needed is for the markers to highlight in UI (if we go that way at all).